### PR TITLE
feat: 🎸 withdraw affirmation for an instruction

### PR DESCRIPTION
**Pull request by sansan**
### JIRA Link 

https://polymesh.atlassian.net/browse/DA-72

### Changelog / Description 

- adds an endpoint to withdraw an affirmation for an isntruction

### Checklist - 

- [x] New Feature ?
- [x] Updated swagger annotation (if API structure is changed) ?
- [x] Unit Test (if possible) ?
- [x] Updated the Readme.md (if required) ?

### CODE REVIEW

File: src/settlements/settlements.controller.ts [hunk 1]
```diff
@@ -133,6 +133,31 @@ export class SettlementsController {
...
+  @ApiParam({
+    name: 'id',
+    description: 'The ID of the Instruction from which to withdraw the affirmation',
+    type: 'string',
+    example: '123',
+  })
+  @ApiOkResponse({
+    description: 'Details of the transaction',
+    type: TransactionQueueModel,
+  })
```
Comment (prashantasdeveloper) [inline src/settlements/settlements.controller.ts:150]: Add ApiNotFoundResponse as well for the case where no instruction is found for given ID

**Review by prashantasdeveloper — CHANGES_REQUESTED** [review]

**Review by prashantasdeveloper — APPROVED** [review]

File: src/settlements/settlements.service.spec.ts [hunk 2]
```diff
@@ -424,7 +424,7 @@ describe('SettlementsService', () => {
   });
 
   describe('rescheduleInstruction', () => {
-    it('should run a reschedule procedure and return the queue data', async () => {
+    it('should run a withdraw affirmation procedure and return the queue data', async () => {
```
Comment (polymath-eric) [inline src/settlements/settlements.service.spec.ts:427]: line shouldn't be changed

**Review by polymath-eric — APPROVED** [review]

**Pull request merged**